src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorEndPointObjectCommand.cs

```diff
@@ -0,0 +1,161 @@
+                }
+                else if (!string.IsNullOrEmpty(this.Address))
+                {
+                    EndpointName = this.Address;
+                }
+            }
+
+            PSNetworkWatcherConnectionMonitorEndpointObject endPoint = new PSNetworkWatcherConnectionMonitorEndpointObject()
+            {
+                Name = string.IsNullOrEmpty(this.Name)? EndpointName : this.Name,
+                ResourceId = this.ResourceId,
+                Address = this.Address,
+            };
+
+            if (this.FilterItem != null)
+            {
+                endPoint.Filter = new PSConnectionMonitorEndpointFilter()
+                {
+                    Type = this.FilterType == null ? "Include" : this.FilterType
+                };
+
+                foreach (PSConnectionMonitorEndpointFilterItem Item in this.FilterItem)
+                {
+                    if (endPoint.Filter.Items == null)
+                    {
+                        endPoint.Filter.Items = new List<PSConnectionMonitorEndpointFilterItem>();
+                    }
+
+                    endPoint.Filter.Items.Add(new PSConnectionMonitorEndpointFilterItem()
+                    {
+                        Type = string.IsNullOrEmpty(Item.Type) ? "AgentAddress" : Item.Type,
+                        Address = Item.Address
+                    });
+                }
+            }
+
+            WriteObject(endPoint);
+        }
+
+        public bool Validate()
+        {
+            if (string.IsNullOrEmpty(this.ResourceId) && string.IsNullOrEmpty(this.Address) && string.IsNullOrEmpty(this.FilterType) && this.FilterItem == null)
+            {
+                throw new ArgumentException("No Parameter is provided");
+            }
+
+            if (string.IsNullOrEmpty(this.ResourceId) && string.IsNullOrEmpty(this.Address))
+            {
+                throw new ArgumentException("ResourceId or Address can not be both empty");
+            }
+
+            if (!string.IsNullOrEmpty(this.ResourceId) && !string.IsNullOrEmpty(this.Address))
+            {
+                throw new ArgumentException("Endpoint in connection monitor should have either ResourceId or Address. These parameters should not be specified together.");
+            }
+
+            if (!string.IsNullOrEmpty(ResourceId))
+            {
+                string[] SplittedName = ResourceId.Split('/');
+
+                // Resource ID must be in this format
+                // "resourceId": "/subscriptions/96e68903-0a56-4819-9987-8d08ad6a1f99/resourceGroups/MyResourceGroup/providers/Microsoft.Compute/virtualMachines/iraVmTest2"
+                if (SplittedName.Count() < 9)
+                {
+                    throw new ArgumentException("ResourceId not in the correct format");
+                }
+            }
+
+            if (!string.IsNullOrEmpty(this.FilterType) && String.Compare(this.FilterType, "Include", true) != 0)
```

Comment:
This should be provided through a ValidateSet - again, to allow the user to tab through allowed values